### PR TITLE
Avocado-Sales-API-Express-Trista

### PR DESCRIPTION
**Pull request by trista1987**
## Render link
https://avocado-express-api.onrender.com

## Collaborators
Solo

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,29 +1,70 @@
...
+    res.json(avocado)
+  } else {
+    res.status (404).send ('Avocado is not found')
+  }
+})
+
+app.get("/date", (req, res) => {
+  const avocadoDate = req.query.date
+
+  let newAvocado = [...avocadoData]
```
Comment (El1an3) [inline server.js:39]: I like the use of the shallow copy here

File: server.js [hunk 2]
```diff
@@ -1,29 +1,70 @@
...
+    res.json(newAvocado)
+  } else{
+    res.status(404).send ('Avocado is not found')
+  }})
+
+  
+app.get("/region", (req, res) => {
+  const regionName = req.query.region
+
+  let countryName = [...avocadoData]
```
Comment (El1an3) [inline server.js:54]: and here ;-)

**Review by El1an3 — COMMENTED** [review]
Well done! Your code meets the requirements as far as I can see. It's harder to give a code review this week, as we all did more or less the same. :-)

File: server.js [hunk 3]
```diff
@@ -1,29 +1,70 @@
...
+  let newAvocado = [...avocadoData]
+
+  if(avocadoDate) {
+    newAvocado = newAvocado.filter(item => item.date.includes(avocadoDate))
+  } 
+  if(!avocadoDate){
+    res.json(newAvocado)
+  } else{
+    res.status(404).send ('Avocado is not found')
+  }})
```
Comment (AntonellaMorittu) [inline server.js:48]: remove this endpoint 🧹

File: server.js [hunk 4]
```diff
@@ -1,29 +1,70 @@
...
+    countryName = countryName.filter(item => item.region.toLowerCase().includes(regionName.toLowerCase()))
+  }
+
+  if(regionName.length>0){
+    res.json(countryName)
+  } else {
+    res.status(404).send ('Avocado is not found')
+  }
+
+})
```
Comment (AntonellaMorittu) [inline server.js:66]: remove this endpoint 🧹

File: server.js [hunk 5]
```diff
@@ -1,29 +1,70 @@
...
 app.use(cors());
 app.use(express.json());
 
 // Start defining your routes here
+
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  const endpoints = expressListEndpoints(app);
+  res.json(endpoints)
+})
```
Comment (AntonellaMorittu) [inline server.js:18]: when you list the endpoints you can add more info to document your API, for example you can filter the data by queries like `region` or `date`, you could add something like this:

```javascript
app.get("/", (req, res) => {
  const endpoints = expressListEndpoints(app);

  // Update documentation for the /avocados endpoint
  const avocadosEndpoint = endpoints.find(endpoint => endpoint.path === '/avocados');
  if (avocadosEndpoint) {
    avocadosEndpoint.queryParams = {
      region: "Filter by region (case-insensitive).",
      date: "Filter by date."
    };
  }

  res.json(endpoints);
});
``` 

this way you keep the usage of your API well documented ⭐

**Review by AntonellaMorittu — APPROVED** [review]
Hi Trista, well done completing your first API! 👏 
Your code indeed meets the specified requirements, nice! 😎  I left comments to improve your endpoints and documentation ⭐

**Pull request closed without merging**